### PR TITLE
[1LP][RFR] This PR will automate SSA on EC2 instance.

### PR DESCRIPTION
**Pull request by sbulage**
PRT run:
{{ pytest: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py -k test_ssa_vm -vvvv --long-running --use-provider ec2west }}

### CODE REVIEW

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 1]
```diff
@@ -164,6 +164,25 @@ def local_setup_provider(request, setup_provider_modscope, provider, appliance):
         request.addfinalizer(appliance.uninstall_vddk)
         set_hosts_credentials(appliance, request, provider)
 
+    if provider.one_of(EC2Provider):
+        version = appliance.version.vstring
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Required: Add request.addfinalizer to clear docker creds

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 2]
```diff
@@ -155,6 +155,35 @@ def _hosts_remove_creds():
...
+                              "docker_registry": "docker.io"}}}}
+    appliance.update_advanced_settings(agent_data)
+
+    # Adding SmartState Docker credentials
+    data = provider.data.endpoints
+    username = credentials[data['ssa_agent']['credentials']]['username']
+    password = credentials[data['ssa_agent']['credentials']]['password']
+    view = navigate_to(provider, "Edit")
+    ssa_agent_view = view.browser.create_view(EC2EndpointForm)
+    ssa_agent_view.smart_state_docker.select()
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Required:
ssa_agent_view.smart_state_docker.fill({
'username', username,
'password': password
})

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 3]
```diff
@@ -155,6 +155,35 @@ def _hosts_remove_creds():
...
+    ssa_agent_view.smart_state_docker.username.fill(username)
+    ssa_agent_view.smart_state_docker.password.fill(password)
+    ssa_agent_view.default.validate.click()
+    view.save.click()
+
+    @request.addfinalizer
+    def _remove_docker_creds():
+        view = navigate_to(provider, "Edit")
+        ssa_agent_view = view.browser.create_view(EC2EndpointForm)
+        ssa_agent_view.smart_state_docker.select()
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: Required:
ssa_agent_view.smart_state_docker.fill({
'username', ''
})

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: okay will add it.

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py (hunk 3 above)
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: okay will add it.

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: @rlbabyuk Adding fill method suggested above, not adding creds as required. reverting back to filling them one after another.

File: cfme/tests/cloud_infra_common/test_vm_instance_analysis.py [hunk 4]
```diff
@@ -155,6 +155,32 @@ def _hosts_remove_creds():
...
+    data = provider.data.endpoints
+    username = credentials[data['ssa_agent']['credentials']]['username']
+    password = credentials[data['ssa_agent']['credentials']]['password']
+    view = navigate_to(provider, "Edit")
+    ssa_agent_view = view.browser.create_view(EC2EndpointForm)
+    ssa_agent_view.smart_state_docker.fill({'username': username, 'password': password})
+    ssa_agent_view.default.validate.click()
+    view.save.click()
+
+    @request.addfinalizer
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_vm_instance_analysis.py]: :+1:

**Pull request merged**